src/core/manager.c

```diff
@@ -3482,29 +3485,52 @@ int manager_set_default_rlimits(Manager *m, struct rlimit **default_rlimit) {
         return 0;
 }
 
-void manager_recheck_journal(Manager *m) {
+static bool manager_journal_is_running(Manager *m) {
         Unit *u;
 
         assert(m);
 
+        /* If we are the user manager we can safely assume that the journal is up */
```

Comment:
I think this is unreachable because of the pid 1 check in the caller.

Reply:
hmm, so yes, that's kinda true, but I wanted to keep this generically valid...

There's a slight different between checking MANAGER_IS_SYSTEM() and getpid() == 1. The former might hold when people call "systemd --system --test" or within "systemd-analyze verify --system" and various test cases like that. The latter only holds when we are actually live, as PID 1.

Now, I figured a function that answers whether journald.service/journald.socket are up should operate in those test modes, too correctly, because there's no reason to break it in that case. However, the call that actually switches around the logging subsystem only really makes sense when we *are* PID 1, as in those tests tools we actually want all output to go to stdout/stderr and not the journal.

hence i conditionalized the logging target switching on a PID 1 check, while the raw check whether journal is up checks whether we are in system scope. because the latter is only

Now, if the getpid() ==1 check holds, then MANAGER_IS_SYSTEM() always holds too, but not vice versa. hence logically you are right, we'll never call manager_journal_is_running()  unless MANAGER_IS_SYSTEM() is also true, however since the tests are slightly different still, and because manager_journal_is_running() might be useful in unrelated cases too one day (it could be as simple as including the result in manager_dump() output for debugging purposes) I opted to keep them in, even though strictly redundant.

Reply:
Sorry, I was not explicit enough.

My real concern is that manager_recheck_journal() will exit immediately if it is not a system instance.  Therefore, in the user manager case we will never get as far as `log_set_prohibit_ipc(true)`.manager case.

Reply:
...and context.  Sorry, I see the original `log_set_prohibit_ipc(false)` is guarded by the same check for PID 1.

Reply:
yeah, the idea is that the manager won't ever touch the prohibit_ipc at all here unless we are PID 1... it won't even set it to true initially...